packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -355,6 +310,8 @@ function parseChildren(
     parse5ParentLocation: parse5.ElementLocation
 ): void {
     const children = (parse5Utils.getTemplateContent(parse5Parent) ?? parse5Parent).childNodes;
+
+    ctx.beginSiblingScope();
```

Comment:
I refactored the sibling tracking into the context object so we shouldn't need to have manual logic here anymore.